core-ui/src/main/java/com/anytypeio/anytype/core_ui/widgets/objectIcon/AvatarIconView.kt

```diff
@@ -0,0 +1,114 @@
+package com.anytypeio.anytype.core_ui.widgets.objectIcon
+
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.size
+import androidx.compose.foundation.shape.CircleShape
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.colorResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.unit.Dp
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import com.anytypeio.anytype.core_ui.R
+import com.anytypeio.anytype.core_ui.common.DefaultPreviews
+import com.anytypeio.anytype.core_ui.views.AvatarTitle
+import com.anytypeio.anytype.presentation.objects.ObjectIcon
+
+
+val avatarBackgroundColor = R.color.shape_tertiary
+val avatarTextColor = R.color.glyph_active
+
+@Composable
+fun AvatarIconView(
+    modifier: Modifier,
+    iconSize: Dp,
+    icon: ObjectIcon.Profile.Avatar,
+    isCircleShape: Boolean = true
+) {
+    val iconParams = getAvatarIconParams(iconSize)
+
+    Box(
+        modifier = modifier
+            .size(iconSize)
+            .background(
+                shape = if (isCircleShape) {
+                    CircleShape
+                } else {
+                    RoundedCornerShape(iconParams.first.dp)
+                },
+                color = colorResource(id = avatarBackgroundColor)
+            ),
+        contentAlignment = Alignment.Center
+    ) {
+        Text(
+            text = icon
+                .name
+                .ifEmpty { stringResource(id = R.string.u) }
+                .take(1)
+                .uppercase(),
+            style = AvatarTitle,
+            fontSize = iconParams.second.sp,
+            color = colorResource(id = avatarTextColor)
+        )
+    }
+}
+
+private fun getAvatarIconParams(size: Dp): Pair<Int, Int> {
```

Comment:
not clear what int means here. should we use some type alias for the sake of alias ?